File: qt/gen_qshareddata.cpp
// +build ignore

#include <QSharedData>
#include <qshareddata.h>
#include "gen_qshareddata.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<_GUID*>( self );
	} else {
		delete self;
	}
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}

QSharedData* QSharedData_new() {
	return new QSharedData();
}

QSharedData* QSharedData_new2(QSharedData* param1) {
	return new QSharedData(*param1);
}

void QSharedData_Delete(QSharedData* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QSharedData*>( self );
	} else {
		delete self;
	}
}

QAdoptSharedDataTag* QAdoptSharedDataTag_new() {
	return new QAdoptSharedDataTag();
}

void QAdoptSharedDataTag_Delete(QAdoptSharedDataTag* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QAdoptSharedDataTag*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qshareddata.go
package qt

import (
	"unsafe"
)

type QSharedData struct {
	h          uintptr
	isSubclass bool
}

// NewQSharedData constructs a new QSharedData object.
func NewQSharedData() *QSharedData {

	ret := newQSharedData(QSharedData_new())
	ret.isSubclass = true
	return ret
}

// NewQSharedData2 constructs a new QSharedData object.
func NewQSharedData2(param1 *QSharedData) *QSharedData {

	ret := newQSharedData(QSharedData_new2(param1.cPointer()))
	ret.isSubclass = true
	return ret
}

type QAdoptSharedDataTag struct {
	h          uintptr
	isSubclass bool
}

// NewQAdoptSharedDataTag constructs a new QAdoptSharedDataTag object.
func NewQAdoptSharedDataTag() *QAdoptSharedDataTag {

	ret := newQAdoptSharedDataTag(QAdoptSharedDataTag_new())
	ret.isSubclass = true
	return ret
}
